include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_banded.hpp

```diff
@@ -0,0 +1,200 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_single_column_banded.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/band/static_band.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_base.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_proxy_type.hpp>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
+#include <seqan3/std/span>
+
+namespace seqan3::detail
+{
+
+/*!\brief A banded alignment score matrix storing only a single banded column for the computation.
```

Comment:
Wait... you are misusing the term **score matrix** here. 
As a score matrix we understand something like BLOSUM62 or a position-dependent score matrix. You mean the dp matrix that contains the prefix scores up to the current position. Please think about renaming the class and file name here and in related files.